### PR TITLE
Use make() to initialize String in NewString for a better performance in sets package

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Use make() to initialize String in NewString for better performance. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/sets/string.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type String map[string]Empty
 
 // NewString creates a String from a list of values.
 func NewString(items ...string) String {
-	ss := String{}
+	ss := make(String, len(items))
```
Comment (ash2k) [inline staging/src/k8s.io/apimachinery/pkg/util/sets/string.go:31]: Could do the same optimization on line 40

Comment (sanposhiho) [reply staging/src/k8s.io/apimachinery/pkg/util/sets/string.go:31]: Sure. Updated.

**Comment by ash2k** [comment]
Ah, I realized that these sets are all generated code so doesn't make sense to change them directly. This is the [generator entry point](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/set-gen/main.go), it invokes the [set generator](https://github.com/kubernetes/gengo/blob/master/examples/set-gen/generators/sets.go). You need to update it there, [bump the dependency](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/vendor.md), and regenerate all the sets. Then ALL of them will get the optimization.

**Comment by sanposhiho** [comment]
>  these sets are all generated code

@ash2k Oh, okay thanks. will close this PR.

/close

**Pull request closed without merging**